### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vsupruniuk**

### CODE REVIEW

**Review by illiarerun — CHANGES_REQUESTED** [review]
Gj ,now let's work on some fixes:

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,37 @@
...
+        for (const key in actions[i].keysToRemove) {
+          delete stateCopy[actions[i].keysToRemove[key]];
+        }
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
```
[the remark this answers is not in the document]
Comment (vsupruniuk) [reply src/transformStateWithClones.js:31]: Unfortunately, I have no idea how to do this😢

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const value of actions) {
+    const { type, extraData, keysToRemove } = value;
```
Comment (jstmpelowycz) [inline src/transformStateWithClones.js:14]: 🔥

**Review by jstmpelowycz — APPROVED** [review]
awesome 🚀

**Pull request closed without merging**